lib/publiccloud/provider.pm

```diff
@@ -337,8 +389,23 @@ Destroys the current terraform deployment
 sub terraform_destroy {
     my ($self) = @_;
     record_info('INFO', 'Removing terraform plan...');
-    assert_script_run('cd ' . TERRAFORM_DIR);
-    script_run('terraform destroy -no-color -auto-approve', TERRAFORM_TIMEOUT);
+    if (get_var('PUBLIC_CLOUD_SLES4SAP')) {
+        assert_script_run('cd ' . TERRAFORM_DIR . '/' . $self->conv_openqa_tf_name);
+    }
+    else {
+        assert_script_run('cd ' . TERRAFORM_DIR);
+    }
+    assert_script_run('terraform plan -no-color -out mydestroy', TERRAFORM_TIMEOUT);
+    my $ret = script_run('terraform apply -no-color -auto-approve mydestroy', TERRAFORM_TIMEOUT);
+    unless (defined $ret) {
+        type_string(qq(\c\\));        # Send QUIT signal
+        assert_script_run('true');    # make sure we have a prompt
+        record_info('ERROR', 'Terraform destroy failed with timeout', result => 'fail');
+        assert_script_run('cd ' . TERRAFORM_DIR);
+        $self->on_terraform_timeout();
```

Comment:
I guess we need to distinguish between `destroy` timeout and `apply` timeout.

Reply:
I don't understand why?

Reply:
The only case where we use this callback is https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/publiccloud/azure.pm#L186 and I wouldn't try to get diagnostics on cleanup, or?

Reply:
Yes that's a good point: better to have access to be able to diagnose? Or better to be sure that VMs are destroyed? From my/our (team) point of view it's better to ensure that VMs are destroyed (still the costs issue!).

Reply:
Yes same from our side!
Side note to https://publiccloud.qa.suse.de/ this is how we catch up left overs.